src/legacy/core_plugins/kibana/public/management/sections/index_patterns/edit_index_pattern/create_edit_field/create_edit_field.tsx

```diff
@@ -24,8 +24,8 @@ import { FieldEditor } from 'ui/field_editor';
 import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
 import { HttpStart, DocLinksStart } from 'src/core/public';
+import { IndexPattern, DataPublicPluginStart } from 'src/plugins/data/public';
```

Comment:
Of course I could 🙂 
But before changing this, I'd like to know details why it is necessary?
AFAIK the usage of `I` prefixes is no longer necessary since we update typescript to 3.7 (this thing was discussed on kibana app migration meeting, @flash1293 please correct me if I'm wrong).
If it's still relevant, then we should get rid of exporting the `IndexPattern` from the data start and export the only interface, because `IndexPattern` continues to be used as interface in many places.

Reply:
In addition, changing this to `IIndexPattern` will cause a lot of changes in data plugin itself, since the Field class constructor expecs the `IndexPattern` type: https://github.com/elastic/kibana/blob/master/src/plugins/data/public/index_patterns/fields/field.ts#L53
the same is about `FieldList`:
https://github.com/elastic/kibana/blob/master/src/plugins/data/public/index_patterns/fields/field_list.ts#L49
and this chain is quite big :)

Reply:
It's not urgent.
`IIndexPattern` is actually a narrower interface than `IndexPattern`
We will take care of that in another iteration. :+1: